feat(Misskey): Add Misskey

ssmucny opened the pull request:
**Description**
Adding chart for Misskey in incubator. Misskey is a fediverse microblogging social media platform.

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [x] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [x] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [x] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [x] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

ssmucny committed:
Importing original chart files for misskey

ssmucny committed:
update chart.yaml

ssmucny committed:
Adding truecharts common library

ssmucny committed:
Merge branch 'truecharts:master' into misskey

ssmucny committed:
debugging setup

ssmucny committed:
Merge branch 'truecharts:master' into misskey

ssmucny committed:
Merge branch 'truecharts:master' into install

ssmucny committed:
Edits to password management

ssmucny committed:
Merge branch 'install' into misskey

ssmucny committed:
updated TrueNAS questions for misskey

ssmucny committed:
Update to Chart.yaml

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/truecharts/charts?pullRequest=4315) <br/>All committers have signed the CLA.

xstar97 commented on charts/incubator/misskey/Chart.yaml:
```diff
@@ -0,0 +1,41 @@
+  truecharts.org/catagories: |
+    - Social
+type: application
+version: 0.1.0
```
chart version should start at `0.0.1`

xstar97 commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,163 @@
+image:
+  repository: misskey/misskey
+  pullPolicy: IfNotPresent
+  tag: latest
```
we recommend not using latest tags if versioning scheme is created.

for example use this until our mirror is created if you made the pr in the `containers` repo: 

```yaml
image:
  repository: misskey/misskey
  tag: 12.119.0@sha256:a7beecb59b5089068463ca3847ac34dcb8445108da75ed8f56822f670053aa31
  pullPolicy: IfNotPresent
```

xstar97 replied:
just copy this into `values.yaml` and replace your image section.

`@sha256:a7beecb59b5089068463ca3847ac34dcb8445108da75ed8f56822f670053aa31` is **required**.

ssmucny replied:
PR for container https://github.com/truecharts/containers/pull/10812. Will update to tccr.io when merged.

ssmucny committed:
Fix chart version

Signed-off-by: Sam Smucny <smucny.sam@gmail.com>

xstar97 commented on charts/incubator/misskey/questions.yaml:
```diff
@@ -0,0 +1,155 @@
+                              required: true
+# Include{advancedPortHTTP}
+# Include{serviceExpertRoot}
+            default: false
```
looks like your missing the target port here. grant it we have plenty of charts that use `3000`, maybe try using the next variable [port](https://truecharts.org/docs/charts/default_port_list)? unless this service needs to be exposed on`3000`, thats fine.
```yaml
                    - variable: main
                      label: Main Service Port Configuration
                      schema:
                        additional_attrs: true
                        type: dict
                        attrs:
                          - variable: port
                            label: Port
                            description: This port exposes the container port on the service
                            schema:
                              type: int
                              default: 3000
                              required: true
# Include{advancedPortHTTP}
                                - variable: targetPort
                                  label: Target Port
                                  description: The internal(!) port on the container the Application runs on
                                  schema:
                                    type: int
                                    default: 3000
# Include{serviceExpertRoot}
            default: false
```

ssmucny replied:
Added targetPort and changed the default ports to 3003.

ssmucny committed:
Add target port UI question

ssmucny committed:
Cleanup and comment values.yaml

xstar97 commented on charts/incubator/misskey/questions.yaml:
```diff
@@ -0,0 +1,161 @@
+                                  description: "The internal(!) port on the container the Application runs on"
+                                  schema:
+                                    type: int
+                                    default: 3003
```
Don't change the target port... leave it as `3000` as its the port used internally.

ssmucny committed:
Addressing review comments

stavros-k commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,159 @@
+hostAliases:
+  - ip: '{{ .Values.env.AccessIP | default "127.0.0.1" }}'
+    hostnames:
+      - "{{ if .Values.ingress.main.enabled }}{{ with (first .Values.ingress.main.hosts) }}{{ .host }}{{ end }}{{ else }}placeholder.fakedomain.dns{{ end }}"
```
Why is that needed?

stavros-k commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,159 @@
+  runAsGroup: 0
+  fsGroup: 33
+
+
```
```suggestion
```

stavros-k commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,159 @@
+  pullPolicy: IfNotPresent
+  tag: 12.119.0@sha256:a7beecb59b5089068463ca3847ac34dcb8445108da75ed8f56822f670053aa31
+
+
```
```suggestion

```

stavros-k commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,159 @@
+  enabled: true
+
+hpb:
+  enabled: false
```
What does this do?

Crow-Control replied:
Nothing, he just blindly copied nextcloud

stavros-k commented on charts/incubator/misskey/templates/ConfigMap.yaml:
```diff
@@ -0,0 +1,172 @@
+    #   You need to set Certificate in 'https' section.
+
+    # To use option 1, uncomment below line.
+    port: {{ .Values.service.main.ports.main.targetPort }} # A port that your Misskey server should listen.
```
```suggestion
    port: {{ .Values.service.main.ports.main.port }} # A port that your Misskey server should listen.
```

This will make it both easier for Scale users to change port (target port is double hidden, port not) 
and you can remove targetPort from values.yaml (it will default to `port`

ssmucny replied:
Thanks for the suggestion.

stavros-k commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,159 @@
+    ports:
+      main:
+        port: 3000
+        targetPort: 3000
```
```suggestion
        port: 3003
```
Generally values on questions and values.yaml should be in sync
(unless it's needed to be different)

stavros-k commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,159 @@
+            echo "Changing ownership to group 33..."
+            chown -R :33 "/misskey/files"
+          fi
+          chmod 770 /misskey/files
```
User has to be able to opt **in** for auto perms.
We have already globally disabled automperms by default (expect VPN config files on addons)

Crow-Control replied:
He blindly copied Nextcloud, which is not a format that should be used by just any App

stavros-k commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,159 @@
+        sed -i "s/POSTGRES_HOST/$POSTGRES_HOST/" /misskey/.config/default.yml
+        sed -i "s/POSTGRES_PASSWORD/$POSTGRES_PASSWORD/" /misskey/.config/default.yml
+        sed -i "s/REDIS_HOST/$REDIS_HOST/" /misskey/.config/default.yml
+        sed -i "s/REDIS_PASSWORD/$REDIS_HOST_PASSWORD/" /misskey/.config/default.yml
```
Why don't you prepare the configmap directly and just mount it in the container?
There is no need for sed here.

ssmucny replied:
Upstream Misskey currently reads in password and host data through the config map yaml file. Since as far as I know, to protect secrets, Kubernetes does not allow you to embed raw secrets in the deployment files. To get them in the config file, I used an init container to read the generated secrets from the env variables and write to the mounted config file. Open to suggestions if there is a better method you are aware of.

stavros-k replied:
You can pull values from dependencies like redis/pg and put them in the configmap.
Search for `.Values.redis.redisPassword` for examples

You can also consider put the whole config in a secret, instead of a configmap

ssmucny replied:
I found redisPassword and postgresqlPassword. For the host names, is <appname>-redis and <appname>-postgresql considered stable, or is the host name generated using a different method?

stavros-k replied:
Yes those are stable, you can use this:
`{{ printf "%v-%v" .Release.Name "postgresql" }}` 
`{{ printf "%v-%v" .Release.Name "redis" }}`

stavros-k committed:
Apply suggestions from code review

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

Crow-Control commented on charts/incubator/misskey/templates/ConfigMap.yaml:
```diff
@@ -0,0 +1,172 @@
+apiVersion: v1
+kind: ConfigMap
```
Configmap needs to be in tpl format, loaded in common.yaml after the init of values.

Crow-Control commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,156 @@
+          else
+            echo "Changing ownership to group 33..."
+            chown -R :33 "/misskey/files"
+          fi
```
We do NOT want permission management outside of Common, unless me or @stavros-k sign off on it.

ssmucny replied:
Will remove

ssmucny committed:
Addressing PR Reviews:
- remove targetPort (only using port)
- update image src to tccr.io
- remove vestigial nextcloud values
- remove auto file permissions

stavros-k commented on charts/incubator/misskey/values.yaml:
```diff
@@ -0,0 +1,129 @@
+  repository: tccr.io/misskey/misskey
+  pullPolicy: IfNotPresent
+  tag: 12.119.0@sha256:a7beecb59b5089068463ca3847ac34dcb8445108da75ed8f56822f670053aa31
+securityContext:
```
```suggestion
image:
  repository: tccr.io/truecharts/misskey
  pullPolicy: IfNotPresent
  tag: 12.119.0@sha256:e16467a28e7cee4442e29216a292dd725f28c3789fb1da050359c7842c2c0eec
  
securityContext:
```

stavros-k commented on charts/incubator/misskey/questions.yaml:
```diff
@@ -66,12 +66,6 @@ questions:
-                                  description: "The internal(!) port on the container the Application runs on"
-                                  schema:
-                                    type: int
-                                    default: 3000
```
This has to stay in questions tho. :)

(Removing target port from values.yaml is ok, because it gets calculated later on from the common library.)

ssmucny replied:
I see, I misunderstood you earlier comment about where it it auto-generated from.

stavros-k replied:
Cool, np!

ssmucny committed:
Update charts/incubator/misskey/values.yaml

The hash value from truecharts is different than docker.io

Co-authored-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>
Signed-off-by: Sam Smucny <smucny.sam@gmail.com>

ssmucny committed:
Add back targetPort in questions.yaml

ssmucny committed:
Move configmap to tpl format and include in common

ssmucny committed:
Merge branch 'master' into misskey

ssmucny committed:
Move secrets into configfile templates:
- remove initcontainer
- update host/pass values in default.yml
- remove misskeyconfig-src pvc
- remove secret env vars

stavros-k commented on charts/incubator/misskey/templates/_configmap.tpl:
```diff
@@ -65,15 +65,15 @@ data:
-      user: "{{ .Values.postgresql.postgresqlUsername }}"
-      pass: "POSTGRES_PASSWORD"
+      user: {{ .Values.postgresql.postgresqlUsername }}
+      pass: {{ .Values.postgresql.postgresqlPassword }}
```
add the trimAll on both passwords

ssmucny committed:
trim " from passwords

ssmucny committed:
Add other misskey configuration options

ssmucny committed:
Merge branch 'master' into misskey

ssmucny committed:
typo

ssmucny committed:
Merge branch 'master' into misskey

ssmucny committed:
Fix indentation

stavros-k committed:
adds couple more options

stavros-k committed:
whoops

stavros-k commented:
@all-contributors please add @ssmucny for code

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
